pkg/runtime/wfengine/backend.go

```diff
@@ -0,0 +1,267 @@
+	select {
+	case <-ctx.Done():
+		return ctx.Err()
+	case be.activityWorkItemChan <- wi:
+		return nil
+	}
+}
+
+// ScheduleWorkflow implements workflowScheduler
+func (be *actorBackend) ScheduleWorkflow(ctx context.Context, wi *backend.OrchestrationWorkItem) error {
+	select {
+	case <-ctx.Done():
+		return ctx.Err()
+	case be.orchestrationWorkItemChan <- wi:
+		return nil
+	}
+}
+
+// CreateOrchestrationInstance implements backend.Backend and creates a new workflow instance.
+//
+// Internally, creating a workflow instance also creates a new actor with the same ID. The create
+// request is saved into the actor's "inbox" and then executed via a reminder thread. If the app is
+// scaled out across multiple replicas, the actor might get assigned to a replicas other than this one.
+func (be *actorBackend) CreateOrchestrationInstance(ctx context.Context, e *backend.HistoryEvent) error {
+	if err := be.validateConfiguration(); err != nil {
+		return err
+	}
+
+	var workflowInstanceID string
+	if es := e.GetExecutionStarted(); es == nil {
+		return errors.New("the history event must be an ExecutionStartedEvent")
+	} else if oi := es.GetOrchestrationInstance(); oi == nil {
+		return errors.New("the ExecutionStartedEvent did not contain orchestration instance information")
+	} else {
+		workflowInstanceID = oi.GetInstanceId()
+	}
+
+	eventData, err := backend.MarshalHistoryEvent(e)
+	if err != nil {
+		return err
+	}
+
+	// Invoke the well-known workflow actor directly, which will be created by this invocation
+	// request. Note that this request goes directly to the actor runtime, bypassing the API layer.
+	req := invokev1.
+		NewInvokeMethodRequest(CreateWorkflowInstanceMethod).
+		WithActor(WorkflowActorType, workflowInstanceID).
+		WithRawDataBytes(eventData).
+		WithContentType(invokev1.OctetStreamContentType)
+	defer req.Close()
+
+	resp, err := be.actors.Call(ctx, req)
+	if err != nil {
+		return err
+	}
+	defer resp.Close()
+	return nil
+}
+
+// GetOrchestrationMetadata implements backend.Backend
+func (be *actorBackend) GetOrchestrationMetadata(ctx context.Context, id api.InstanceID) (*api.OrchestrationMetadata, error) {
+	// Invoke the corresponding actor, which internally stores its own workflow metadata
+	req := invokev1.
+		NewInvokeMethodRequest(GetWorkflowMetadataMethod).
+		WithActor(WorkflowActorType, string(id)).
+		WithContentType(invokev1.OctetStreamContentType)
+	defer req.Close()
+
+	if res, err := be.actors.Call(ctx, req); err != nil {
+		return nil, err
+	} else {
+		defer res.Close()
+		data, err := res.RawDataFull()
+		if err != nil {
+			return nil, fmt.Errorf("failed to read the internal actor response: %w", err)
+		}
+		if len(data) == 0 {
+			return nil, api.ErrInstanceNotFound
+		}
+		var metadata api.OrchestrationMetadata
+		if err := actors.DecodeInternalActorData(data, &metadata); err != nil {
+			return nil, fmt.Errorf("failed to decode the internal actor response: %w", err)
+		}
+		return &metadata, nil
+	}
```

Comment:
nit: typically this else is omitted following the go code style

Reply:
Will fix this.